### ISSUE TITLE
Creating a yolo model with input size 1920 1080

### ISSUE THREAD

**Issue by thegkhn** [ultralytics/ultralytics#4526]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

`!python train.py --batch 8 --img 1920 --epochs 200--cfg /content/drive/MyDrive/AI/yolov7/cfg/training/yolov7.yaml --hyp /content/drive/MyDrive/AI/yolov7/data/hyp.scratch.p5.yaml --data /content/drive/MyDrive/AI/ndatalast/a/data.yaml --weights 'yolov7_training.pt'`

I create a model using the code above, but the performance is extremely bad. But when I create this model in 640*640 size, that is, when I update the --img-size parameter to 640, the model runs successfully. I have done a lot of research on this subject on the internet. When I use the --rect parameter, the result is very bad. 
My question is this:
Is it possible to create the model with 1080 width and 1920 height input shape? At the same time, I am training this model on colab regardless of performance, so the resource is not much of a problem for me. So the only thing I am trying to do is to train a model with 1080 width and 1920 height input shape. Can I do this in yolov7 and yolov8 model? Thanks in advance for your answers.

**Comment by glenn-jocher** [comment]
@thegkhn hi there,

Yes, you absolutely can train a YOLOv8 model with an input shape of 1080 width and 1920 height. The `--img` command you're currently using sets both dimensions of the training images to the specified size, essentially creating a square. Therefore, if you want a rectangular shape like 1080x1920, you will need to modify the pre-processing code to resize the images while preserving these dimensions.

Please keep in mind that using a higher resolution like 1080x1920 will increase the demand for resources, even though you mentioned that this isn't an issue for you on Colab. High-resolution images have more pixels, which means the model will take longer to process and may require more memory.

As for the `--rect` command, it indicates to use rectangular training. This means it maintains the aspect ratio of the loaded images and doesn't reshape them into squares. However, model performance can vary with rectangular training, and it's not always the best option depending on the dataset and the specific task.

Training on higher-resolution images can sometimes improve performance, but it can also make training more challenging. If you’re experiencing bad results, it might not solely be due to the input size but could be influenced by other factors, such as the architecture of the model, learning rate, number of epochs, etc.

Feel free to play around with these parameters and see which work best for your specific use case. Remember to monitor your model's performance throughout the training process to ensure that it is learning effectively.

**Comment by thegkhn** [comment]
> @thegkhn hi there,
> 
> Yes, you absolutely can train a YOLOv8 model with an input shape of 1080 width and 1920 height. The `--img` command you're currently using sets both dimensions of the training images to the specified size, essentially creating a square. Therefore, if you want a rectangular shape like 1080x1920, you will need to modify the pre-processing code to resize the images while preserving these dimensions.
> 
> Please keep in mind that using a higher resolution like 1080x1920 will increase the demand for resources, even though you mentioned that this isn't an issue for you on Colab. High-resolution images have more pixels, which means the model will take longer to process and may require more memory.
> 
> As for the `--rect` command, it indicates to use rectangular training. This means it maintains the aspect ratio of the loaded images and doesn't reshape them into squares. However, model performance can vary with rectangular training, and it's not always the best option depending on the dataset and the specific task.
> 
> Training on higher-resolution images can sometimes improve performance, but it can also make training more challenging. If you’re experiencing bad results, it might not solely be due to the input size but could be influenced by other factors, such as the architecture of the model, learning rate, number of epochs, etc.
> 
> Feel free to play around with these parameters and see which work best for your specific use case. Remember to monitor your model's performance throughout the training process to ensure that it is learning effectively.
> 

Thank you very much for your reply. Do you have any suggestion for improving performance for high resolutions photo like this?

**Comment by glenn-jocher** [comment]
@thegkhn training a model with a high input resolution like 1080 width and 1920 height can be challenging. It's often a balance between performance and computation resources. I would like to provide some general suggestions which could help improve your model's performance:

1. **Batch Size**: Increase your batch size if your hardware can handle it. Generally, a larger batch size might improve your model.

2. **Learning Rate**: You might need to fine-tune the learning rate. A high learning rate might cause the model to converge too quickly to a suboptimal solution, while a low learning rate might cause unnecessarily long training time.

3. **Data Augmentation**: This enables the model to see the "same" image in different ways, thereby enriching the learning. However, keep in mind that not all augmentations are useful for all data.

4. **Optimization algorithm**: Experiment with different optimization algorithms like SGD, Adam, etc. Some optimizers might work better than others depending on your specific scenario.

5. **Network Architecture**: If you are using a pre-defined architecture, try to experiment with different architectures which might perform better for your specific task.

6. **Input Resolution**: Although you are advised to train with the same resolution you'll be detecting on for optimal results, this comes with higher computational costs. You can try downsampling your images and analyze the performance.

7. **Feature Extractor**: Try using a different feature extractor or modifying the existing one. This really depends on your specific case, but it might help.

Keep in mind that machine learning involves a lot of experimentations, so don't be afraid to test different configurations until you find one that performs best for your scenario. And remember to monitor your training process to make sure the model is training as expected. 

I hope this helps, and best of luck with your YOLOv8 training!

**Comment by thegkhn** [comment]
> @thegkhn training a model with a high input resolution like 1080 width and 1920 height can be challenging. It's often a balance between performance and computation resources. I would like to provide some general suggestions which could help improve your model's performance:
> 
> 1. **Batch Size**: Increase your batch size if your hardware can handle it. Generally, a larger batch size might improve your model.
> 2. **Learning Rate**: You might need to fine-tune the learning rate. A high learning rate might cause the model to converge too quickly to a suboptimal solution, while a low learning rate might cause unnecessarily long training time.
> 3. **Data Augmentation**: This enables the model to see the "same" image in different ways, thereby enriching the learning. However, keep in mind that not all augmentations are useful for all data.
> 4. **Optimization algorithm**: Experiment with different optimization algorithms like SGD, Adam, etc. Some optimizers might work better than others depending on your specific scenario.
> 5. **Network Architecture**: If you are using a pre-defined architecture, try to experiment with different architectures which might perform better for your specific task.
> 6. **Input Resolution**: Although you are advised to train with the same resolution you'll be detecting on for optimal results, this comes with higher computational costs. You can try downsampling your images and analyze the performance.
> 7. **Feature Extractor**: Try using a different feature extractor or modifying the existing one. This really depends on your specific case, but it might help.
> 
> Keep in mind that machine learning involves a lot of experimentations, so don't be afraid to test different configurations until you find one that performs best for your scenario. And remember to monitor your training process to make sure the model is training as expected.
> 
> I hope this helps, and best of luck with your YOLOv8 training!

Thank you for your answer. I did and my new model is greater than old models.
